### PR TITLE
[KED-914] Dive CLI command

### LINKED ISSUE #8: [KED-914] Image analysis using dive
## Description
Investigate feasibility of [dive](https://github.com/wagoodman/dive) as per @yarncraft's suggestion in #5.

## Context
`dive` can be integrated into CI to perform image default image efficiency.

## Possible Implementation
Integration of `dive` into e2e tests.

### PR DESCRIPTION
**Pull request by DmitriiDeriabinQB**
## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

## Motivation and Context
Resolves #8 

## How has this been tested?
e2e tests

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Assigned myself to the PR
- [x] Added tests to cover my changes
- [x] Added new entries to the `RELEASE.md` file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,6 +104,23 @@ Options:
...
 
+### Image analysis with Dive
+
+Kedro-Docker allows to analyze the size efficiency of your project image by leveraging [Dive](https://github.com/wagoodman/dive):
+
+```bash
+kedro docker dive
+```
+
+> *Note:* By default Kedro-Docker calls Dive in CI mode. If you want to explore your image in the UI, run `kedro docker dive --no-ci`.
```
Comment (921kiyo) [inline README.md]: `kedro docker dive` works fine, but I got an error when I run `kedro docker dive --no-ci` with the following error stack (not sure what's was the problem, either my side or somewhere in the changes). 

```
(docker-dev36) ➜  hello kedro docker dive --no-ci
Traceback (most recent call last):
  File "/Users/kiyohito_kunii/anaconda3/envs/docker-dev36/bin/kedro", line 10, in <module>
    sys.exit(main())
  File "/Users/kiyohito_kunii/anaconda3/envs/docker-dev36/lib/python3.6/site-packages/kedro/cli/cli.py", line 618, in main
    ("Project specific commands", project_groups),
  File "/Users/kiyohito_kunii/anaconda3/envs/docker-dev36/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/kiyohito_kunii/anaconda3/envs/docker-dev36/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/kiyohito_kunii/anaconda3/envs/docker-dev36/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/kiyohito_kunii/anaconda3/envs/docker-dev36/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/kiyohito_kunii/anaconda3/envs/docker-dev36/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/kiyohito_kunii/anaconda3/envs/docker-dev36/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/kiyohito_kunii/anaconda3/envs/docker-dev36/lib/python3.6/site-packages/kedro_docker/plugin.py", line 344, in docker_dive
    required_args=required_args, optional_args=optional_args, user_args=docker_args
  File "/Users/kiyohito_kunii/anaconda3/envs/docker-dev36/lib/python3.6/site-packages/kedro_docker/helpers.py", line 154, in compose_docker_run_args
    for arg_name, arg_value in optional_args:
ValueError: too many values to unpack (expected 2)
```

File: kedro_docker/plugin.py [hunk 2]
```diff
@@ -47,11 +46,12 @@
     make_container_name,
 )
 
+CONTEXT_SETTINGS = dict(help_option_names=["-h", "--help"])
```
Comment (921kiyo) [inline kedro_docker/plugin.py]: Are we planning to add more stuff in this dict in the future so this is more generic like context?

File: README.md (hunk 1 above)
Comment (DmitriiDeriabinQB) [reply README.md]: It was indeed a CLI bug, thanks very much for spotting it 🙏

File: kedro_docker/plugin.py (hunk 2 above)
Comment (DmitriiDeriabinQB) [reply kedro_docker/plugin.py]: I see your point 😉  Removed the constant for now.

File: kedro_docker/plugin.py [hunk 3]
```diff
@@ -298,3 +301,48 @@ def docker_cmd(args, docker_args, image):
...
+    help="Path to `.dive-ci` config file",
+)
+@_make_image_option(callback=_image_callback)
+@_make_docker_args_option()
+def docker_dive(ci_flag, dive_ci, docker_args, image):
+    """Run Dive analyzer of Docker image efficiency."""
+    container_name = make_container_name(image, "dive")
+
+    required_args = [("-v", "/var/run/docker.sock:/var/run/docker.sock")]
+    optional_args = [("--rm", None), ("--name", container_name)]
```
Comment (921kiyo) [inline kedro_docker/plugin.py]: In Dive docs, I can see examples of `--rm -it` together. If cl_flag is not passed, optional_args would be `--rm --name xxx`?

Comment (DmitriiDeriabinQB) [reply kedro_docker/plugin.py]: Yes, correct, `-it` flag is not needed when running Dive in CI mode

File: features/docker.feature [hunk 4]
```diff
@@ -85,11 +85,11 @@ Feature: Docker commands in new projects
     Then I should get an error exit code
     And Standard error should contain a message including "Error: Port 8890 is already in use on the host. Please specify an alternative port number."
 
-  Scenario: Execute docker kedro-test target
+  Scenario: Execute docker kedro test target
     Given I have executed the kedro command "docker build"
     When I execute the kedro command "docker cmd kedro test"
     Then I should get a successful exit code
-    And I should get a message including "3 passed, 1 warnings"
+    And I should get a message including "2 passed"
```
Comment (lorenabalan) [inline features/docker.feature]: Why, what changed?

File: kedro_docker/plugin.py [hunk 5]
```diff
@@ -298,3 +301,48 @@ def docker_cmd(args, docker_args, image):
 
     command = ["docker", "run"] + _docker_run_args + [image] + list(args)
     call(command)
+
+
+@docker_group.command(name="dive")
+@click.option(
+    "--ci/--no-ci",
```
Comment (lorenabalan) [inline kedro_docker/plugin.py]: In the readme it says that it goes to CI by default, is there any point in having that option here then?

File: features/docker.feature (hunk 4 above)
Comment (DmitriiDeriabinQB) [reply features/docker.feature]: That was a change on Kedro side

File: kedro_docker/plugin.py (hunk 5 above)
Comment (DmitriiDeriabinQB) [reply kedro_docker/plugin.py]: I personally prefer this on/off flag as it gives a better code readability and allows to change default behaviour without changing the code. Here is an [example](https://click.palletsprojects.com/en/7.x/options/#boolean-flags) from the docs.

**Comment by yarncraft** [comment]
Nice work!

**Pull request merged**